src/ngx_http_modsecurity_module.c

```diff
@@ -520,6 +539,31 @@ ngx_module_t ngx_http_modsecurity_module = {
 };
 
 
+static ngx_http_variable_t  ngx_http_modsecurity_vars[] = {
+    { ngx_string("modsecurity_req_headers_phase_time"), NULL,
+      ngx_http_modsecurity_req_headers_phase_time, 0,
+      NGX_HTTP_VAR_NOCACHEABLE, 0 },
+
+    { ngx_string("modsecurity_req_body_phase_time"), NULL,
+      ngx_http_modsecurity_req_body_phase_time, 0,
+      NGX_HTTP_VAR_NOCACHEABLE, 0 },
+
+    { ngx_string("modsecurity_resp_headers_phase_time"), NULL,
+      ngx_http_modsecurity_resp_headers_phase_time, 0,
+      NGX_HTTP_VAR_NOCACHEABLE, 0 },
+    
+    { ngx_string("modsecurity_resp_body_phase_time"), NULL,
+      ngx_http_modsecurity_resp_body_phase_time, 0,
+      NGX_HTTP_VAR_NOCACHEABLE, 0 },
+
+    { ngx_string("modsecurity_logging_phase_time"), NULL,
+      ngx_http_modsecurity_logging_phase_time, 0,
```

Comment:
use a same function with different argument, like 0, 1, 2, ...

Reply:
Refer https://github.com/nginx/nginx/blob/release-1.21.6/src/http/ngx_http_upstream.c#L391-L401
Same function `ngx_http_upstream_response_time_variable` for 3 variables.